### DISCUSSION TITLE
Adaptive Mean Thresholding

### DISCUSSION THREAD

**Discussion by e012345678** [SixLabors/ImageSharp#2639] [Q&A]
Does anyone know how to do Adaptive Mean Thresholding like in this link: https://www.tutorialspoint.com/how-to-perform-adaptive-mean-and-gaussian-thresholding-of-an-image-using-python-opencv

My current code is like this:

```
using (var grayImage = dst.CloneAs<L8>())
{
   
    AdaptiveThresholdProcessor adaptiveThresholdProcessor = new AdaptiveThresholdProcessor();
   // How to use AdaptiveThresholdProcessor, should it be used?

    using (var memoryStream = new MemoryStream())
    {
        grayImage.SaveAsPng(memoryStream); 
        byteArray = memoryStream.ToArray();
    }
}
```
Specfically the opencv2 line I am trying to port is: 
`Cv2.AdaptiveThreshold(dst, thresholded, 255, AdaptiveThresholdTypes.MeanC, ThresholdTypes.Binary, 85, 10);`

Any tips on how to do that with ImageSharp?

**Comment by JimBobSquarePants** [comment]
Use the processing APIs

```c#
Image.Mutate( x => x.AdaptiveThreshold())
```

There’s several [overloads](https://docs.sixlabors.com/api/ImageSharp/SixLabors.ImageSharp.Processing.AdaptiveThresholdExtensions.html) available.

**Reply by e012345678** [reply]
Thanks a lot for the reply and thanks for a great lib! I am bit unsure of those parameters and types though. Can I pass MeanC and blocksize as in that line of code I have?
